numba/cuda/cudaimpl.py

```diff
@@ -625,6 +625,40 @@ def ptx_atomic_sub(context, builder, dtype, ptr, val):
         return builder.atomic_rmw('sub', ptr, val, 'monotonic')
 
 
+@lower(stubs.atomic.inc, types.Array, types.intp, types.Any)
+@lower(stubs.atomic.inc, types.Array, types.UniTuple, types.Any)
+@lower(stubs.atomic.inc, types.Array, types.Tuple, types.Any)
+@_atomic_dispatcher
+def ptx_atomic_inc(context, builder, dtype, ptr, val):
+    if dtype == types.uint32:
+        lmod = builder.module
+        return builder.call(nvvmutils.declare_atomic_inc_int32(lmod),
+                            (ptr, val))
+    elif dtype == types.uint64:
+        lmod = builder.module
+        return builder.call(nvvmutils.declare_atomic_inc_int64(lmod),
+                            (ptr, val))
```

Comment:
there's no need to change this, but you can write this sort of thing like:
```python
    if dtype in unsigned_int_numba_types:
        bw = dtype.bitwidth
        lmod = builder.module
        fn = getattr(nvvmutils, f'declare_atomic_inc_int{bw}')
        return builder.call(fn(lmod), (ptr, val))
    else:
        raise TypeError(f'Unimplemented atomic inc with {dtype} array')
```

Reply:
I made this change anyway.